src/Common/src/System/Net/SafeSocketHandle.Windows.cs

```diff
@@ -9,8 +9,13 @@
 
 namespace System.Net.Sockets
 {
-    internal partial class SafeCloseSocket :
-#if DEBUG
+#if SYSTEM_NET_SOCKETS_DLL
+    public
+#else
+    internal
+#endif
+        partial class SafeSocketHandle :
+#if DEBUG && !SYSTEM_NET_SOCKETS_DLL
         DebugSafeHandleMinusOneIsInvalid
 #else
         SafeHandleMinusOneIsInvalid
```

Comment:
Similarly, you should be able to leave all of the base class stuff out of both the .Unix.cs file and the .Windows.cs file, and only have it in the main partial file.